Remove `subarg` argument from `dag` commands and add support for `bundle-name` argument

ambika-garg opened the pull request:
<!--
 Licensed to the Apache Software Foundation (ASF) under one
 or more contributor license agreements.  See the NOTICE file
 distributed with this work for additional information
 regarding copyright ownership.  The ASF licenses this file
 to you under the Apache License, Version 2.0 (the
 "License"); you may not use this file except in compliance
 with the License.  You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing,
 software distributed under the License is distributed on an
 "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 KIND, either express or implied.  See the License for the
 specific language governing permissions and limitations
 under the License.
 -->

<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of an existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->
This PR aims to update all DAG commands by removing support for the `subdir` parameter and replacing it with support for the `bundle-name` argument.

related: #45647

ambika-garg committed:
Support --bundle-name in list-import-error command

ambika-garg committed:
Update args of list-import-errors command

ambika-garg committed:
Remove subdir for dags state command

ambika-garg committed:
Support for bundle name in dags report command

jedcunningham commented on airflow/cli/commands/remote_commands/dag_command.py:
```diff
@@ -404,7 +419,21 @@ def dag_list_import_errors(args) -> None:
+        if bundle.name not in bundles_to_reserialize:
+            continue
+        bundle.initialize()
+        dagbag = DagBag(bundle.path, include_examples=False)
```
As written, `dagbag` will only have stats for the last bundle in the loop.

ambika-garg replied:
I apologize, missed it. I have fixed it in the latest commit.

ambika-garg replied:
I have a quick question: if a DAG is erroneous, should it be included in the DAG report, like in the case of `bundle-2.py`?
<img width="896" alt="Screenshot 2025-02-24 at 10 38 38 PM" src="https://github.com/user-attachments/assets/929f9f80-2e13-4170-9bdf-2d88e160faa8" />

jedcunningham replied:
What happens in AF2? Seems fine to me though.

ambika-garg replied:
what is AF2?

jedcunningham replied:
Airflow 2

ambika-garg replied:
It gives the same output.

ambika-garg committed:
Fix: dagbag_stats reads all the bundle stats

ambika-garg committed:
support bundle name in dag-show and next-execution command

ambika-garg committed:
Remove unnecessary ARG_SUBDIR parameters

ambika-garg committed:
* Fix: Test for next-execution command
* Remove unnecessary parameter

ambika-garg committed:
Fix tests

ambika-garg committed:
Fix core db tests

jedcunningham committed:
Remove bundle name from `show` and `next-execution`

dag_id is already unique enough, no need to offer filtering by bundle
name for single entity commands.

jedcunningham committed:
Fix `dags test` and fix tests

jedcunningham committed:
Remove random added files

jedcunningham reviewed (approved):
@ambika-garg I came in and finished this off - we are running out of time. Hope you don't mind.

LGTM, but I'll wait for another committer to approve before I merge as I wrote non-trivial stuff in this PR.

jedcunningham commented on airflow/cli/commands/remote_commands/dag_command.py:
```diff
@@ -576,6 +609,29 @@ def _render_dagrun(dr: DagRun) -> dict[str, str]:
 
+def _parse_and_get_dag(dag_id: str) -> DAG | None:
+    """Given a dag_id, determine the bundle and relative fileloc from the db, then parse and return the DAG."""
+    db_dag = get_dag(subdir=None, dag_id=dag_id, from_db=True)
```
It's already checked [here](https://github.com/apache/airflow/blob/a3435f11bbf2c2c487ec1be53e21ab574f119b61/airflow/utils/cli.py#L284), and the typing on `get_dag` does say `DAG`, not `DAG | None`.

ambika-garg committed:
Merge branch 'main' into remove-subarg-dags

ambika-garg commented on airflow/cli/commands/remote_commands/dag_command.py:
```diff
@@ -576,6 +609,29 @@ def _render_dagrun(dr: DagRun) -> dict[str, str]:
     AirflowConsole().print_as(data=dag_runs, output=args.output, mapper=_render_dagrun)
 
 
+def _parse_and_get_dag(dag_id: str) -> DAG | None:
```
Hey @jedcunningham, should we move this function to `cli.py` file, as it might be used in task_command.py as well.

jedcunningham replied:
We can, but let's do it in that other PR if we end up needing to.

pierrejeambrun reviewed (commented):
Should we emit a newsfragment + linter rule for that ?

jedcunningham committed:
Update 66 newsfragment
